File: src/main/java/Collections.java
import java.util.*;

public class Collections {

    public static void main(String[] args) {

        // Java
        List<String> listFruits = Arrays.asList("apple", "coconut",
                "mango", "avocado");
        listFruits.add("cherry");

        for (int i = 0; i < listFruits.size(); i++) {
            String result = String.format("%s at %d",
                    listFruits.get(i), i);
            System.out.println(result);
        }

        for (String value : listFruits) {
            System.out.println(value);
        }

        listFruits.forEach(System.out::println);

        List<String> filterFruits = new ArrayList<>();

        for (int i = 0; i < listFruits.size(); i++) {
            if (listFruits.get(i).startsWith("a")) {
                filterFruits.add(listFruits.get(i));
            }
        }

        System.out.println(filterFruits.toString());
        // output -> [apple, avocado]


    }


}

File: src/main/kotlin/Collections.kt
//fun main(args: Array<String>) {
//
//    // Kotlin
//    val listFruits: List<String> = listOf("apple", "coconut",
//            "mango", "avocado")
//    listFruits.add("cherry") // compile error
//
//    val mutableFruits: MutableList<String> = mutableListOf("apple",
//            "coconut", "mango", "avocado")
//    mutableFruits.add("cherry")
//
//    for (value in mutableFruits) {
//        println(value)
//    }
//
//    for ((index, value) in mutableFruits.withIndex()) {
//        println("$value at $index")
//    }
//
//    // Functional Programming (FP)
//    mutableFruits.forEach { println(it) }
//    mutableFruits.forEachIndexed { index, value -> println("$value at $index") }
//    val filter = mutableFruits.filter { it.startsWith("a") }
//    println(filter) // output -> [apple, avocado]
//
//
//}